### PR TITLE
fix: onScrollDragBegin dismiss keyboard and blur input

### PR DESCRIPTION
**Pull request by gkartalis**
This PR resolves [FX-4628] 

### Description

Makes sure that onScrollDragBegin the keyboard is dismissed and the input is blurred

##### Android

https://www.loom.com/share/ad512fab10bf48509716820c4550a99e

##### iOS

https://user-images.githubusercontent.com/21178754/219061809-1eb35159-8bb9-417e-9cf2-72718f537fe8.MP4

### PR Checklist

- [x] I have tested my changes on **iOS** and **Android**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [x] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- search results onScrollDragBegin dismiss keyboard and blur input - gkartalis

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

### CODE REVIEW

File: src/app/Scenes/Search/components/ElasticSearchResults.tsx [hunk 1]
```diff
@@ -24,6 +25,7 @@ export const SearchResults2: React.FC<SearchResults2Props> = ({ query, selectedP
   const space = useSpace()
   const isAutosuggestModeEnabled = useFeatureFlag("AREnableAutosuggestModeESSearch")
   const flatListRef = useRef<FlatList>(null)
+  const searchProviderValues = useSearchProviderValues(query)
```
Comment (dimatretyak) [inline src/app/Scenes/Search/components/ElasticSearchResults.tsx:28]: **suggestion**: I think it would be better to use `SearchContext` here

Comment (dimatretyak) [reply src/app/Scenes/Search/components/ElasticSearchResults.tsx:28]: `const { inputRef, queryRef } = useContext(SearchContext)`

Comment (gkartalis) [reply src/app/Scenes/Search/components/ElasticSearchResults.tsx:28]: aaaah, makes sense, to not pass the query in

**Review by dblandin — DISMISSED** [review]
Looks so smooth! 💯

File: src/app/Scenes/Search/SearchContext.ts [hunk 2]
```diff
@@ -1,8 +1,8 @@
 import { InputRef } from "palette"
-import { createContext, Ref, useRef } from "react"
+import { createContext, Ref, RefObject, useRef } from "react"
 
 export const SearchContext = createContext<{
-  inputRef: Ref<InputRef>
+  inputRef: RefObject<InputRef>
```
Comment (gkartalis) [inline src/app/Scenes/Search/SearchContext.ts:5]: **bonus:** fixed the type here as well 👍

**Pull request merged**